dht_net.go

```diff
@@ -214,6 +229,9 @@ func (dht *IpfsDHT) sendMessage(ctx context.Context, p peer.ID, pmes *pb.Message
 	}
 
 	if err := ms.SendMessage(ctx, pmes); err != nil {
+		if err == msmux.ErrNotSupported {
```

Comment:
hmmm I think you're right. I'm not sure why these were initially added, but removing peers from the routing table doesn't seem like an **emergency** operation. All these `dht.RoutingTable().Remove` calls should probably be covered by the events.

@Stebalien any thoughts, otherwise I'm just going to remove these.

Reply:
@aarshkshah1992 is there any reason not to leave these in just in case they're useful due to race conditions like you mentioned at https://github.com/libp2p/go-libp2p-kad-dht/pull/472#discussion_r387587221

Reply:
I think we can leave these here. Was just trying to understand why these are here. Will wait for @Stebalien to reply.

Reply:
We should learn (eventually) that the peer is no longer a DHT peer through identify push/delta. However, this is an _extra_ safety mechanism, just in case.